src/data/calculations/CalculationsRepository.ts

```diff
@@ -0,0 +1,90 @@
+import { IStorageRepository } from "../storage/IStorageRepository";
+import {
+    CalculationData,
+    ICalculationsRepository,
+} from "./ICalculationsRepository";
+
+export class CalculationsRepository implements ICalculationsRepository {
+    remoteDataSource: IStorageRepository = IStorageRepository.instance;
+
+    async storeCalculation(calculationData: CalculationData): Promise<void> {
+        try {
+            const oldCalculations = await this.getAllCalculations();
+            const newCalculations = [calculationData, ...oldCalculations];
+            await this.remoteDataSource.set({
+                allCalculations: JSON.stringify(newCalculations),
+            });
+        } catch (e: unknown) {
+            throw Error(e as string);
```

Comment:
Because e is unknown, e.message does not exist.

I don't think typing in catch clauses are allowed yet: [see link](https://github.com/microsoft/TypeScript/issues/42596)